src/universal/modules/teamDashboard/components/ArchiveTeamConfirmation/ArchiveTeamConfirmation.js

```diff
@@ -0,0 +1,102 @@
+import React, {Component, PropTypes} from 'react';
+import {reduxForm, Field} from 'redux-form';
+import InputField from 'universal/components/InputField/InputField';
+import {cashay} from 'cashay';
+import Button from 'universal/components/Button/Button';
+import {css} from 'aphrodite-local-styles/no-important';
+import appTheme from 'universal/styles/theme/appTheme';
+import withStyles from 'universal/styles/withStyles';
+
+const styleThunk = () => ({
+  errorMessage: {
+    color: appTheme.palette.warm
+  }
+});
+
+@withStyles(styleThunk)
+@reduxForm({form: 'archiveTeamConfirmation'})
+export default class ArchiveTeamConfirmation extends Component {
+
+  static propTypes = {
+    teamName: PropTypes.string.isRequired,
+    teamId: PropTypes.string.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    router: PropTypes.object.isRequired,
+    styles: PropTypes.object.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showConfirmationField: false,
+      showError: false
+    };
+  }
+
+  archiveTeam = () => {
+    const {teamName, teamId, router} = this.props;
+    const variables = {
+      updatedTeam: {
+        id: teamId,
+        name: teamName,
+        isArchived: true
+      }
+    };
+    cashay.mutate('archiveTeam', {variables});
+    router.push('/me');
+  }
+
+  formSubmit = (data) => {
+    const {teamName} = this.props;
+    const {archivedTeamName} = data;
+    if (teamName === archivedTeamName) {
+      this.archiveTeam();
+    } else {
+      this.setState({showError: true});
```

Comment:
oh yeah, this is definitely one of those redux-form gotchas.
redux-form already gives us the props to determine this.
first, we'll want to move all of this logic into the form field component, so on line 92 you are passing all that field logic into the `InputField` component.
That component has a line that looks like this:
```
{touched && error && <FieldHelpText hasErrorText helpText={error} />}
```
basically that says if they've clicked inside the input field & there is an error, show the error.
that way, we don't have to mess around with the state, we can do it all from the props it gives us.